packages/react-intersection-observer/README.md

```diff
@@ -0,0 +1,49 @@
+# `@shopify/react-intersection-observer`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-intersection-observer.svg)](https://badge.fury.io/js/%40shopify%2Freact-intersection-observer.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/react-intersection-observer.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/react-intersection-observer.svg)
+
+A simple React wrapper around the [IntersectionObserver API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-intersection-observer
+```
+
+## Usage
+
+This package exports an `IntersectionObserver` component. This component will create its own DOM node that will be observed, with optional `onIntersecting` and `onNotIntersecting` props being called as appropriate. `onIntersecting` and `onNotIntersecting` will be called with an `IntersectionObserverEntry` object, which describes the state of the intersection. The component also accepts a few additional props that correspond to the options used to construct an `IntersectionObserver`:
+
+- `threshold`: a number or array of number indicating the `intersectionRatio` that must be met before the observer is triggered
+- `root`: a string or element that is used as the viewport for visibility testing. If a string is passed, it will be treated as a selector.
+- `rootMargin`: a string representing the margins by which to shrink the root’s bounding box before computing intersections.
+
+This component also allows you to customize the rendered markup. You can pass `children`, which will be rendered inside of the element being observed for intersections. You can also pass an `element` prop that changes the DOM node being wrapped around those children (defaults to a `div`). Regardless of the passed `element`, the `IntersectionObserver` component will always add a `display: contents` style to that element in order to reduce the styling impact of the additional nesting.
```

Comment:
`element` is a confusing name for me.  It's surrounded by references to React/DOM nodes (`element being observed`, `this.parentElement`), but accepts a text value.

`wrapperTagName` is verbose, but descriptive.